server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/VariableWidthHistogramAggregator.java

```diff
@@ -385,8 +385,8 @@ private int getNearestBucket(double value){
             if (numClusters == 0){
                 return -1;
             }
-            BigArrays.DoubleBinarySearcher searcher = new BigArrays.DoubleBinarySearcher(clusterCentroids);
-            return searcher.search(0, numClusters - 1, value);
+            BigArrays.DoubleBinarySearcher binarySearcher = new BigArrays.DoubleBinarySearcher(clusterCentroids);
```

Comment:
I still think it'd be nice to make this one time. I don't think the JVM is going to manage to optimize away this allocation and its fairly important that we build don't allocate more than we have to when we're collecting hits from a single bucket.

For a sort of degenerate example, have a look at #55559. The old way had a bunch of allocations and `instanceofs` on every collect and redoing it without all that made it about 1400% faster. Just allocating here isn't going to be anything like that kind of hit. And, hell, maybe the JVM can optimize away the allocation. I'm not sure to be honest. But if it can't then you'll find the allocation and young generation GC will dominate the collection time.

All of the other allocation that you do is all fine because it is amortized. That is why stuff like `grow` is so nice - it'll oversize the array so on average we don't allocate on every new bucket.

This is one time where silly java just gets in the way!